lib/mayaUsd/fileio/primUpdaterManager.cpp

```diff
@@ -760,6 +760,14 @@ class PushPullScope
         }
     }
 
+    void end()
+    {
+        if (_controllingFlag) {
+            *_controllingFlag = false;
+            _controllingFlag = nullptr;
+        }
+    }
+
```

Comment:
This allows resetting the "in-push-pull" flag early because otherwise it would interfere with the auto-edit mechanism that is triggered by the final additional command when merging to USD a Maya Ref with auto-edit on.